src/pyhf/modifiers/staterror.py

```diff
@@ -50,47 +51,39 @@ def __init__(self, staterr_mods, pdfconfig, mega_mods):
                 for m in keys
             ]
         )
+        self.finalize(pdfconfig)
 
-        if self._staterror_indices:
-            access_rows = []
-            staterror_mask = default_backend.astensor(self._staterror_mask)
-            for mask, inds in zip(staterror_mask, self._staterror_indices):
-                summed_mask = default_backend.sum(mask[:, 0, :], axis=0)
-                assert default_backend.shape(
-                    summed_mask[summed_mask > 0]
-                ) == default_backend.shape(default_backend.astensor(inds))
-                # make masks of > 0 and == 0
```

Comment:
how did we get rid of the masks? that's amazing...

Reply:
using param_viewer we have better control over which indices we need and we can more explicitly construct the access field as commented on the code